mapper/orm/src/main/java/org/hibernate/search/mapper/orm/session/spi/HibernateOrmSearchManager.java

```diff
@@ -27,4 +28,6 @@
 	 */
 	PojoWorkPlan createWorkPlan();
 
+	SessionWorkExecutor createSessionWorkExecutor();
```

Comment:
Seeing this makes me think the naming is a bit inconsistent. Do you think you could rename `MappingWorkExecutor` to `PojoMappingWorkExecutor` and `SessionWorkExecutor` to `PojoSessionWorkExecutor`? No need to rename the methods (`createSessionWorkExecutor` etc.) or the variables, just renaming the type will be enough.